nucliadb/nucliadb/search/search/shards.py

```diff
@@ -33,35 +33,47 @@
 from nucliadb_protos.noderesources_pb2 import Shard
 
 from nucliadb.ingest.orm.node import Node
+from nucliadb_telemetry import metrics
 
+node_observer = metrics.Observer("node_client", labels={"type": ""})
 
-def query_shard(node: Node, shard: str, query: SearchRequest) -> SearchResponse:
+
+async def query_shard(node: Node, shard: str, query: SearchRequest) -> SearchResponse:
     query.shard = shard
-    return node.reader.Search(query)
+    with node_observer({"type": "search"}):
+        return await node.reader.Search(query)  # type: ignore
```

Comment:
Did this never work before as a coroutine? I wonder if we are using the method at all then

Reply:
Yup because it was still be awaited and the `Search` method still returned an awaitable result. Changing where we await here just allows us to observe the call.